pkg/scheduler/plugins/drf/hdrf_test.go

```diff
@@ -49,82 +52,42 @@ type pgSpec struct {
 }
 
 func TestHDRF(t *testing.T) {
-	klog.InitFlags(nil)
-	flag.Set("v", "4")
-	flag.Set("alsologtostderr", "true")
-	var tmp *cache.SchedulerCache
-	patches := gomonkey.ApplyMethod(reflect.TypeOf(tmp), "AddBindTask", func(scCache *cache.SchedulerCache, task *api.TaskInfo) error {
-		scCache.Binder.Bind(nil, []*api.TaskInfo{task})
-		return nil
-	})
-	defer patches.Reset()
-
-	patchUpdateQueueStatus := gomonkey.ApplyMethod(reflect.TypeOf(tmp), "UpdateQueueStatus", func(scCache *cache.SchedulerCache, queue *api.QueueInfo) error {
-		return nil
-	})
-	defer patchUpdateQueueStatus.Reset()
-
-	s := options.NewServerOption()
-	s.MinNodesToFind = 100
-	s.PercentageOfNodesToFind = 100
-	s.RegisterOptions()
-
-	framework.RegisterPluginBuilder(PluginName, New)
-	framework.RegisterPluginBuilder("proportion", proportion.New)
-	defer framework.CleanupPluginBuilders()
+	options.Default()
 
+	plugins := map[string]framework.PluginBuilder{PluginName: New, proportion.PluginName: proportion.New}
 	tests := []struct {
-		name       string
-		pgSpecs    []pgSpec
-		nodes      []*v1.Node
-		queues     []*schedulingv1.Queue
-		queueSpecs []queueSpec
-		expected   map[string]*api.Resource
+		uthelper.TestCommonStruct
+		expected map[string]*api.Resource
 	}{
 		{
-			name: "rescaling test",
-			pgSpecs: []pgSpec{
-				{
-					taskNum: 10,
-					cpu:     "1",
-					mem:     "1G",
-					pg:      "pg1",
-					queue:   "root-sci",
-				},
-				{
-					taskNum: 10,
-					cpu:     "1",
-					mem:     "0G",
-					pg:      "pg21",
-					queue:   "root-eng-dev",
-				},
-				{
-					taskNum: 10,
-					cpu:     "0",
-					mem:     "1G",
-					pg:      "pg22",
-					queue:   "root-eng-prod",
-				},
-			},
-			nodes: []*v1.Node{util.BuildNode("n",
-				api.BuildResourceList("10", "10G", []api.ScalarResource{{Name: "pods", Value: "50"}}...),
-				make(map[string]string))},
-			queueSpecs: []queueSpec{
-				{
-					name:      "root-sci",
-					hierarchy: "root/sci",
-					weights:   "100/50",
-				},
-				{
-					name:      "root-eng-dev",
-					hierarchy: "root/eng/dev",
-					weights:   "100/50/50",
-				},
-				{
-					name:      "root-eng-prod",
-					hierarchy: "root/eng/prod",
-					weights:   "100/50/50",
-				},
+			TestCommonStruct: uthelper.TestCommonStruct{
+				Name:    "rescaling test",
+				Plugins: plugins,
+				PodGroups: []*schedulingv1.PodGroup{
+					util.BuildPodGroup("pg1", "default", "root-sci", 0, nil, schedulingv1.PodGroupInqueue),
+					util.BuildPodGroup("pg21", "default", "root-eng-dev", 0, nil, schedulingv1.PodGroupInqueue),
+					util.BuildPodGroup("pg22", "default", "root-eng-prod", 0, nil, schedulingv1.PodGroupInqueue),
+				},
+				Pods: mergePods(
```

Comment:
Is this method truly needed? We can declare a slice directly.

Reply:
How？each makePods returns a list, and I need to merge them to one list. Append looks more complex

Reply:
OK